src/HermesSerializers/defautSerializer.js

```diff
@@ -15,100 +12,98 @@ module.exports =  {
-                for (let k = 0; k < keys.length; k++) {
-                    obj.value[keys[k]] = args[i][keys[k]];
-                }
-                serializedArgs.push(obj);
-            } else if (args[i] instanceof DataView) {
-                transferable.push(args[i].buffer);
-                serializedArgs.push({
-                    type: "DataView",
-                    value: args[i].buffer,
+
+                const keys = Object.getOwnPropertyNames(argument);
+                keys.reduce((acc, key) => {
+                    acc[key] = argument[key];
+                    return acc;
+                }, obj.value);
+
+                acc.serializedArgs.push(obj);
+                return acc;
+            }
+
+            if (argument instanceof ArrayBuffer) {
+                acc.transferable.push(argument);
+                acc.serializedArgs.push({
+                    type: "__hermes__transferable__",
+                    className: "ArrayBuffer",
+                    index: acc.transferable.length - 1,
                 });
-            } else {
-                // transferable: ArrayBuffer
-                if (args[i] instanceof ArrayBuffer) {
-                    transferable.push(args[i]);
-                    args[i] = "hermes__transferable__" + transferable.length - 1;
+                return acc;
+            }
 
-                // tranferable: ImageData
-                } else if ("ImageData" in self && args[i] instanceof ImageData) {
-                    transferable.push(args[i].data.buffer);
-                    args[i] = {
-                        type: "hermes__transferable__ImageData",
-                        index: transferable.length - 1,
-                        width: args[i].width,
-                    };
-                // tranferable: TypedArray
-                } else {
-                    for (let t = 0; t < typedArray.length; t++) {
-                        if (args[i] instanceof self[typedArray[t]]) {
-                            transferable.push(args[i].buffer);
-                            args[i] = "hermes__transferable__" + transferable.length - 1;
-                            break;
-                        }
-                    }
-                }
+            if (argument instanceof ImageData && "ImageData" in self) {
+                acc.transferable.push(argument.data.buffer);
+                acc.serializedArgs.push({
+                    type: "__hermes__transferable__",
+                    className: "ImageData",
+                    index: acc.transferable.length - 1,
+                    data: {
+                        width: argument.width,
+                    },
+                });
+                return acc;
+            }
 
-                serializedArgs.push({
-                    type: "arg",
-                    value: args[i],
+            if (argument && typeof argument === "object" && argument.constructor.name in self && typedArray.includes(argument.constructor.name)) {
```

Comment:
could it be useful to test `argument.constructor` as well ?

Reply:
It's not necessaries because all objects in JavaScript have an attribute constructor (We have checked type of argument is object)

https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Global_Objects/Object/constructor